RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Text.RegularExpressions;
+
+namespace RummikubGameApp
+{
+    public class GameLogic
+    {
+        private Tile[] tiles;
+        public static Tile[] hand;
+        private Tile[] AIhand;
+        private Tile[,] deck;
+        private int currentPlayerIndex;
+        private int tilesIndex;

```

Comment:
the names are problematic, tiles is the actual deck while the deck is the board. should change tiles to deck and deck to board